x/ibc/03-connection/keeper/handshake.go

```diff
@@ -120,7 +121,7 @@ func (k Keeper) ConnOpenTry(
 		bytes.Equal(previousConnection.Counterparty.Prefix.Bytes(), counterparty.Prefix.Bytes()) &&
```

Comment:
nit: I think the previousConnection check should happen way earlier, before the version gets picked and current connection is created.

Reply:
I agree but this follows the ICS spec ordering so I think it makes sense to leave it where it is so its easier to verify correctness of the implementation against the spec. I'd suggest reordering this on the spec before reordering this in the code

Reply:
nit for @cwgoes